astropy/coordinates/attributes.py

```diff
@@ -106,32 +105,8 @@ def __get__(self, instance, frame_cls=None):
                 out = getattr(instance, self.secondary_attribute, self.default)
 
         out, converted = self.convert_input(out)
-        if instance is not None:
```

Comment:
Hmm, this looks wrong. Why is this dead code? Any time an attribute is gotten from `instance`, this should be exercised (instance can be `None` only if one gets the attribute from the class, like `FK4.equinox`).

And the code does something useful -- it broadcasts the attribute as need be. Which really should be used... (why did I forget this? I fear it was me who wrote this...)

Reply:
It's dead code because:

- When BaseFrame.__init__ calls getattr to convert the attributes, it is NOT doing the broadcasting; it _can't_ do the broadcasting at that point because it hasn't yet computed the desired final shape.
- After determining the final shape, then BaseFrame.__init__ loops over the attributes and broadcasts each of them.
- Consequently, after BaseFrame.__init__ has finished, _all_ of the attributes have been broadcast.
- When you get attributes, this code is never reached, because they broadcasting was done in advance.

Reply:
OK, I see what you did. However, this may be slightly less performant because now we are potentially doing the type conversion twice.

Reply:
Looking at this code again, I'm indeed not sure it is the `Attribute` that should be doing the broadcasting -- I don't like that I have to write a comment that `getattr(self, key)` does the broadcasting, bad code smell! But I think that refactoring is probably best done separately from this PR.

Reply:
🥳 Woohoo! What a tricky PR. Thanks for all your feedback. Should I open a follow-up issue for this item now?

Reply:
A follow up issue never hurts, if you have not opened one already. Thank you!